platform/iphone/in_app_store.mm

```diff
@@ -70,42 +50,38 @@ - (NSString *)localizedPrice {
 	ClassDB::bind_method(D_METHOD("pop_pending_event"), &InAppStore::pop_pending_event);
 	ClassDB::bind_method(D_METHOD("finish_transaction"), &InAppStore::finish_transaction);
 	ClassDB::bind_method(D_METHOD("set_auto_finish_transaction"), &InAppStore::set_auto_finish_transaction);
-};
+}
 
-@interface ProductsDelegate : NSObject <SKProductsRequestDelegate> {
-};
+@interface ProductsDelegate : NSObject <SKProductsRequestDelegate, SKPaymentTransactionObserver>
```

Comment:
This object is now functioning as both the products request delegate and as a transaction observer. This allowed us to ditch the `TransObserver` class entirely.